src/Uno.UI/UI/Xaml/Input/FocusManager.wasm.cs

```diff
@@ -40,9 +42,28 @@ internal static void ProcessElementFocused(UIElement element)
 				_log.Value.LogDebug($"{nameof(ProcessElementFocused)}() focusedElement={GetFocusedElement()}, element={element}, searching for focusable parent control");
 			}
 
-			// Try to find the first focusable parent and set it as focused, otherwise just keep it for reference (GetFocusedElement())
-			var ownerControl = element.GetParents().OfType<Control>().Where(control => control.IsFocusable).FirstOrDefault();
-			ProcessControlFocused(ownerControl);
+			foreach (var parent in element.GetParents())
+			{
+				_log.Value.LogDebug($"Checking {parent}. TextBlock.IsFocusable: {(parent as TextBlock)?.IsFocusable == true}");
+
+				// Try to find the first focusable parent and set it as focused, otherwise just keep it for reference (GetFocusedElement())
+				if (parent is TextBlock textBlock && textBlock.IsFocusable)
+				{
+					// Focusable TextBlock parent, we can move focus to it.
+					var focusManager = VisualTree.GetFocusManagerForElement(textBlock);
```

Comment:
Just a question: We can have more than one `FocusManager` per `WIndow` ?

Reply:
We don't, but we should in the future 😀 . With proper window management in place, each "window" (I think `CoreWindow` for UWP, some-kind-of-window for WinUI) has its own `FocusManager` "instance" associated with the visual tree.